services/workflows-service/src/workflow/hook-callback-handler.service.ts

```diff
@@ -43,17 +43,7 @@ export class HookCallbackHandlerService {
       );
     }
 
-    set(workflowRuntime.context, resultDestinationPath, data);
-
-    await this.workflowService.updateWorkflowRuntimeData(
-      workflowRuntime.id,
-      {
-        context: workflowRuntime.context,
-      },
-      currentProjectId,
-    );
-
-    return data;
+    return set({}, resultDestinationPath, data);
```

Comment:
Here we just return the new context instead of directly update it on the DB. Now the update happens in the service.